### PR TITLE
Move sigmoid to math_functions

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description

Moving sigmoid to its own file

### Testing Done

made sure everything compiled and all tests built/ran 

### Resolved Issues
#520 

### Length Justification
n/a

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/util/math_functions.cpp [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // Created by roark on 28/03/19.
 //
 
+#include "util/math_functions.h"
+
 #include <algorithm>
 
 namespace Util
```
Comment (garethellis0) [inline src/thunderbots/software/util/math_functions.cpp:9]: Could we leave this outside of the `Util` namespace for now? I'd like to see this functions in the same namespace as the geom stuff, which is just no namespace for now (we're for sure going to re-evaluate all the namespaces after competition).

**Review by garethellis0 — CHANGES_REQUESTED** [review]
Could we make sure to move the unit tests for these functions to a corresponding file as well (making sure the `test` folder structure matches)?

File: src/thunderbots/software/util/math_functions.cpp (hunk 1 above)
Comment (akhilveeraghanta) [reply src/thunderbots/software/util/math_functions.cpp:9]: moved

**Review by garethellis0 — APPROVED** [review]

**Comment by garethellis0** [comment]
LGTM :+1: :tada:

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Pull request merged**